src/utils/jss/ConstructableStyleSheetsRenderer.ts

```diff
@@ -41,19 +41,18 @@ function getStyleSheet(x: HTMLStyleElement) {
     return y
 }
 
-let rafAwaiting = false
 export function applyAdoptedStyleSheets(shadowOnly = true) {
-    if (rafAwaiting) return
-    rafAwaiting = true
     requestAnimationFrame(() => {
-        rafAwaiting = false
-        const styles = Array.from(fakeHead.children).filter((x): x is HTMLStyleElement => x instanceof HTMLStyleElement)
+        const styles = Array.from(fakeHead.children).filter(
+            (x): x is HTMLStyleElement => x instanceof HTMLStyleElement && x.innerText !== '',
+        )
         const shadows = Array.from(livingShadowRoots)
         const nextAdoptedStyleSheets = styles.map(getStyleSheet)
         for (const shadow of shadows) {
-            // Maybe we can remove this now.
             try {
-                shadow.adoptedStyleSheets = nextAdoptedStyleSheets
+                if (shadow.adoptedStyleSheets.length !== nextAdoptedStyleSheets.length) {
```

Comment:
Comparing the length may bring inconsistency in the styling. Changes on CSSStyleSheets (not only add or remove) will also invoke this function too

Reply:
Ah... I have another idea... can we merge these styles into one element... to improve performance?

Reply:
CSSStyleSheet object has internal state. Some methods depends on the order of the CSS rule. If you merge them, you have to maintain the order relationship.

Theoretically check .innerText also brings inconsistency. You should also check if there is any "rules" in the object (the innerText does not sync with the internal "rules")